src/neptune/objects/run.py

```diff
@@ -406,6 +407,9 @@ def __init__(
         if mode == Mode.OFFLINE or mode == Mode.DEBUG:
             project = OFFLINE_PROJECT_QUALIFIED_NAME
 
+        if self._custom_run_id is None and mode != Mode.READ_ONLY:
```

Comment:
why when in read only custom run id is not generated?

Reply:
We want to generate it only if a run is actually created - refer to the task description